src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -216,6 +216,16 @@ readPlanToCountQuery (Node ReadPlan{from=mainQi, fromAlias=tblAlias, where_=logi
       if joinType == Just JTInner
         then ("EXISTS (" <> readPlanToCountQuery readReq <> " )"):rest
         else rest
+    pgFmtLogicTree' qi' (CoercibleExpr hasNot op forest') = SQL.sql notOp <> " (" <> intercalateSnippet (opSql op) (pgFmtLogicTree' qi' <$> forest') <> ")"
+      where
+        notOp =  if hasNot then "NOT" else mempty
+
+        opSql And = " AND "
+        opSql Or  = " OR "
+    pgFmtLogicTree' _ (CoercibleStmnt (CoercibleFilterNullEmbed hasNot fld)) =
+      "(" <> maybe "" readPlanToCountQuery (nullEmbedRel fld) <> ") IS " <> (if hasNot then "NOT" else mempty) <> " NULL"
+    pgFmtLogicTree' qi' stmnt = pgFmtLogicTree qi' stmnt
+    nullEmbedRel fld = find (\(Node ReadPlan{relAggAlias} _) -> fld == relAggAlias) forest
```

Comment:
:open_mouth: Oh, I thought it would be a more involved change. Can you show the resulting query?

Reply:
The full resulting query is like this:

```bash
$ curl "http://localhost:3000/films?select=*,actors(*)&actors=not.is.null"
```

```sql
-- not.is.null

WITH pgrst_source AS
  (SELECT "api"."films".*,
          COALESCE("films_actors_1"."films_actors_1", '[]') AS "actors"
   FROM "api"."films"
   LEFT JOIN LATERAL
     (SELECT json_agg("films_actors_1") AS "films_actors_1"
      FROM
        (SELECT "api"."actors".*
         FROM "api"."actors",
              "api"."roles"
         WHERE "api"."roles"."actor_id" = "api"."actors"."id"
           AND "api"."roles"."film_id" = "api"."films"."id" LIMIT $1
         OFFSET $2) AS "films_actors_1") AS "films_actors_1" ON TRUE
   WHERE "films_actors_1" IS NOT NULL LIMIT $3
   OFFSET $4) ,
pgrst_source_count AS
  (SELECT 1
   FROM "api"."films"
   WHERE
       (SELECT 1
        FROM "api"."actors",
             "api"."roles"
        WHERE "api"."roles"."actor_id" = "api"."actors"."id"
          AND "api"."roles"."film_id" = "api"."films"."id") IS NOT NULL)
SELECT
  (SELECT pg_catalog.count(*)
   FROM pgrst_source_count) AS total_result_set,
       pg_catalog.count(_postgrest_t) AS page_total,
       coalesce(json_agg(_postgrest_t), '[]') AS body,
       nullif(current_setting('response.headers', TRUE), '') AS response_headers,
       nullif(current_setting('response.status', TRUE), '') AS response_status
FROM
  (SELECT *
   FROM pgrst_source) _postgrest_t
```

Reply:
The `pgrst_source_count` is the relevant part. I used `IS NOT NULL` instead of `EXISTS` because I wanted to handle the edge case mentioned here: https://github.com/PostgREST/postgrest/issues/2800.

But this does not work for that case, because the `SELECT 1` will never be `NOT NULL`, we need the full `SELECT` to verify that.

In any case, we need to decide whether to translate `table=not.is.null` to `NOT (table IS NULL)` or go with what we're using now: `table IS NOT NULL`. This comment by Wolfgang in the same issue gives a better context: https://github.com/PostgREST/postgrest/issues/2800#issuecomment-1562620508

Reply:
Now that #2800 is completed, the new query can now use `EXISTS` for `not.is.null` and `NOT EXISTS` for `is.null`. The resulting query (the `pgrst_source_count`) now would be:

```sql
  WITH pgrst_source AS
  (SELECT "api"."films".*,
          COALESCE("films_actors_1"."films_actors_1", '[]') AS "actors"
   FROM "api"."films"
   LEFT JOIN LATERAL
     (SELECT json_agg("films_actors_1") AS "films_actors_1"
      FROM
        (SELECT "api"."actors".*
         FROM "api"."actors",
              "api"."roles"
         WHERE "api"."roles"."actor_id" = "api"."actors"."id"
           AND "api"."roles"."film_id" = "api"."films"."id" LIMIT $1
         OFFSET $2) AS "films_actors_1") AS "films_actors_1" ON TRUE
   WHERE "films_actors_1" IS DISTINCT FROM NULL LIMIT $3
   OFFSET $4) ,
     pgrst_source_count AS
  (SELECT 1
   FROM "api"."films"
   WHERE EXISTS
       (SELECT 1
        FROM "api"."actors",
             "api"."roles"
        WHERE "api"."roles"."actor_id" = "api"."actors"."id"
          AND "api"."roles"."film_id" = "api"."films"."id"))
SELECT
  (SELECT pg_catalog.count(*)
   FROM pgrst_source_count) AS total_result_set,
       pg_catalog.count(_postgrest_t) AS page_total,
       coalesce(json_agg(_postgrest_t), '[]') AS body,
       nullif(current_setting('response.headers', TRUE), '') AS response_headers,
       nullif(current_setting('response.status', TRUE), '') AS response_status
FROM
  (SELECT *
   FROM pgrst_source) _postgrest_t
```